### PR TITLE
perf(install): parallel importer manifest transforms and one-pass link rendering

### PR DESCRIPTION
**Pull request by zkochan**
## Summary

Related to [pnpm/pnpm#14352](https://github.com/pnpm/pnpm/issues/14352) (continuing the large-workspace performance effort).

Re-profiling the warm lockfile-only update on the 6,833-project repro (current `main`, ~1.07 s median locally) surfaced two per-importer costs still paid serially:

1. **`build_manifest_transforms` transformed every importer manifest on the main thread.** The compatibility-DB package extender is active by default, so a workspace-scale install always clones and rewrites all 6,833 importer manifests in a serial loop before resolution starts. Each importer's transform is independent, so the loop now fans out across rayon; the rebuilt `BTreeMap` restores ordering regardless of completion order. Profile: 16 → 2 samples.

2. **`ImporterAnchor::target_relative_to_importer` split the target string twice per edge** — once to find the shared prefix with the importer suffix, and again to render the kept tail (`str::split` was the single largest self-time entry in the profile). One pass over the target's bytes now classifies raw segments, grows the shared prefix, and remembers where the kept tail starts; when the tail is verbatim (no collapsed separators, `.` segments, or trailing separator — the overwhelmingly common case) the rendering is a single `String` build from the climb count and the raw tail slice. A non-verbatim tail falls back to the segment join. New unit tests pin that messy spellings (`packages//lib`, `packages/./lib`, `packages/lib/`, `./packages/lib`) render identically to their clean form, alongside the existing `diff_paths` parity test. Profile: peers pass 56 → 42 samples inclusive.

Benchmark (protocol from the prior PRs in this effort: restore `package.json` + `pnpm-lock.yaml`, flip one root dep `workspace:*` → `workspace:^`, `pnpm install --lockfile-only --trust-lockfile --offline --ignore-scripts`, medians of 10 runs):

| | median | min |
|---|---:|---:|
| main | 1067 ms | 1035 ms |
| this PR | 969 ms | 941 ms |

The produced lockfile is byte-identical: diffing against the committed lockfile shows exactly the one flipped specifier line.

This is pacquet-internal performance work with no user-visible behavior change, so no TypeScript-side change is needed.

## Squash Commit Body

```text
Two costs the warm lockfile-only update profile on the 6,833-project
workspace repro still charged per importer:

- build_manifest_transforms cloned and transformed every importer
  manifest serially; the compatibility-DB extender is active by
  default, so a workspace-scale install always paid the whole loop on
  the main thread. Each importer's transform is independent, so the
  loop now fans out across rayon and the rebuilt BTreeMap restores the
  ordering (16 -> 2 profile samples).

- ImporterAnchor::target_relative_to_importer split the target string
  twice per edge: once to find the shared prefix, and again to render
  the kept tail. One pass over the target's bytes now classifies the
  raw segments, grows the shared prefix, and remembers where the kept
  tail starts; when the tail is verbatim (no collapsed separators, dot
  segments, or trailing separator - the overwhelmingly common case)
  the rendering is a single String build from the climb and the raw
  tail slice. A non-verbatim tail falls back to the segment join, with
  new unit tests pinning that both spellings render identically.

Warm lockfile-only update on the repro: median 1067 ms -> 969 ms, with
the produced lockfile byte-identical (only the flipped specifier
changes). Part of the pnpm/pnpm#14352 performance effort.
```

## Checklist

- [x] I checked the referenced issue and verified that none of the PRs
  already linked to it solves it.
- [x] The change is implemented in both the TypeScript CLI and the Rust
  `pnpm/` port, or the description notes what still needs porting.
  (pacquet-internal performance work; no behavior change to mirror.)
- [x] Added a changeset (`pnpm changeset`) if this PR changes any published
  package. Keep it short and written for pnpm users — it becomes a release note.
- [x] Added or updated tests.

---
Written by an agent (Claude Code, claude-fable-5).

---
<a href="https://app.blacksmith.sh/pnpm/codesmith/pnpm/pr/14523"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with [code]smith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1791078476&installation_model_id=426870&pr_number=14523&repository=pnpm%2Fpnpm&return_to=https%3A%2F%2Fgithub.com%2Fpnpm%2Fpnpm%2Fpull%2F14523&signature=f5bb905a3a185bc0a43313f2a1327117fd43502d36da98b36a865772d50d1366"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with [code]smith</picture></a>
<sup>Need help on this PR? Tag <code>@codesmith-bot</code> with what you need. Autofix is disabled.</sup>

## Summary by CodeRabbit

- **Performance**
  - Dependency resolution now prepares project manifests in parallel, improving installation efficiency while preserving deterministic results.

- **Bug Fixes**
  - Workspace link paths are rendered more reliably, including paths with repeated separators, relative segments, trailing separators, and platform-specific formats.
  - Invalid anchored or drive-prefixed paths are handled safely.

- **Tests**
  - Added coverage for normalized workspace link rendering and edge cases involving equivalent or parent paths.

### CODE REVIEW

File: .changeset/parallel-manifest-transforms.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"pacquet": patch
+---
+
+Sped up dependency resolution in large workspaces. pnpm now prepares project manifests for resolution in parallel and renders workspace link paths in a single pass [#14352](https://github.com/pnpm/pnpm/issues/14352).
```
[the remark this answers is not in the document]
Comment (zkochan) [reply .changeset/parallel-manifest-transforms.md]: Applied in 16668fce6f — the changeset is now the single user-facing sentence.

---
Written by an agent (Claude Code, claude-fable-5).

File: pnpm/crates/resolving-deps-resolver/src/link_target.rs [hunk 2]
```diff
@@ -58,23 +58,78 @@ impl ImporterAnchor {
...
-        let segments = relative_segments(target)?;
+        if target.starts_with(SEPARATORS) {
+            return None;
+        }
+        // One pass over the target's bytes: classify each raw segment,
+        // grow the shared prefix, and remember where the kept tail
+        // starts. The kept tail is almost always verbatim — no
+        // collapsed separators, no `.` segments, no trailing separator
+        // — and then the rendering is one `String` build from two
+        // slices instead of a second segment split.
```
[the remark this answers is not in the document]
Comment (zkochan) [reply pnpm/crates/resolving-deps-resolver/src/link_target.rs]: Trimmed in 16668fce6f: the comment now states only the why (skip the second split) and the invariant that gates the fast path (a verbatim tail), not the loop's steps.

---
Written by an agent (Claude Code, claude-fable-5).
